docs/notebooks/WMD_tutorial.ipynb

```diff
@@ -0,0 +1,603 @@
+    "\n",
+    "Below a JSON file with Yelp reviews is read line by line, the text is extracted, tokenized, and stopwords and punctuation are removed.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[nltk_data] Downloading package punkt to /home/olavur/nltk_data...\n",
+      "[nltk_data]   Package punkt is already up-to-date!\n",
+      "\n",
+      "Cell took 39.26 seconds to run.\n"
+     ]
+    }
+   ],
+   "source": [
+    "start = time()\n",
+    "\n",
+    "import json\n",
+    "from nltk import word_tokenize\n",
+    "download('punkt')  # Download data for tokenizer.\n",
+    "\n",
+    "# Business IDs of some restaurants.\n",
+    "ids = ['4bEjOyTaDG24SY5TxsaUNQ', '2e2e7WgqU1BnpxmQL5jbfw', 'zt1TpTuJ6y9n551sw9TaEg',\n",
+    "      'Xhg93cMdemu5pAMkDoEdtQ', 'sIyHTizqAiGu12XMLX3N3g', 'YNQgak-ZLtYJQxlDwN-qIg']\n",
+    "\n",
+    "# Load some text data.\n",
+    "w2v_corpus = []  # Documents to train word2vec on.\n",
+    "wmd_corpus = []  # Documents to run queries against.\n",
+    "documents = []  # wmd_corpus, with no pre-processing (so we can print the original documents).\n",
+    "with open('/home/olavur/yelp_dataset_challenge_academic_dataset/yelp_academic_dataset_review.json') as data_file:\n",
+    "    for line in data_file:\n",
+    "        json_line = json.loads(line)\n",
+    "        \n",
+    "        if json_line['business_id'] not in ids:\n",
+    "            # Not interested in this business.\n",
+    "            continue\n",
+    "        \n",
+    "        # Pre-process document.\n",
+    "        text = json_line['text'].lower()  # Lower the text.\n",
+    "        text = word_tokenize(text)  # Split into words.\n",
+    "        text = [w for w in text if not w in stop_words]  # Remove stopwords.\n",
+    "        text = [w for w in text if w.isalpha()]  # Remove numbers and punctuation.\n",
+    "        \n",
+    "        # Add to corpus for training Word2Vec.\n",
+    "        w2v_corpus.append(text)\n",
+    "        \n",
+    "        if json_line['business_id'] == ids[0]:\n",
+    "            # Only use this restaurant in corpus for WmdSimilarity.\n",
+    "            wmd_corpus.append(text)\n",
+    "            documents.append(json_line['text'])\n",
+    "\n",
+    "print '\\nCell took %.2f seconds to run.' %(time() - start)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below is a plot with a histogram of document lengths and includes the average document length as well. Note that these are the pre-processed documents, meaning stopwords are removed, punctuation is removed, etc."
```

Comment:
"An average review is a short text of about 60 words as seen on the histogram" . This needs a summary